llvm/lib/Target/DirectX/DXILPrepare.cpp

```diff
@@ -110,9 +140,20 @@ class DXILPrepareModule : public ModulePass {
       if (!isValidForDXIL(I))
         AttrMask.addAttribute(I);
     }
+
+    
+    dxil::ValidatorVersionMD ValVerMD(M);
+    VersionTuple ValVer = ValVerMD.getAsVersionTuple();
+    bool SkipValidation = ValVer.getMajor() == 0 && ValVer.getMinor() == 0;
+
     for (auto &F : M.functions()) {
       F.removeFnAttrs(AttrMask);
       F.removeRetAttrs(AttrMask);
+      // Only remove string attributes if we are not skipping validation.
+      // This will reserve the experimental attributes when validation version
+      // is 0.0 for experiment mode.
+      if (!SkipValidation)
```

Comment:
Do you have an example of an experimental attribute? It seems to me like we don't want to completely disable stripping attributes when validation is skipped.

Reply:
It will be string started with "exp-".
Something like "exp-shader".

Reply:
In that case it seems like we should always strip disallowed attributes (regardless of validation) and just allow experimental attributes when validation is disabled.

Reply:
Updated.